tests/proxy_test.go

```diff
@@ -223,12 +223,12 @@ func runGRPCProxyServer() (proxy, func(), error) {
 	return p, cleanup, err
 }
 
-func runGRPCProxyServerWithServerCount(serverCount int) (proxy, *agentserver.ProxyServer, func(), error) {
+func runGRPCProxyServerWithServerCount(serverCount int) (proxy, *server.ProxyServer, func(), error) {
 	var proxy proxy
 	var err error
 	var lis, lis2 net.Listener
 
-	server := agentserver.NewProxyServer(uuid.New().String(), serverCount, &agentserver.AgentTokenAuthenticationOptions{})
+	server := server.NewProxyServer(uuid.New().String(), serverCount, &server.AgentTokenAuthenticationOptions{})
```

Comment:
nit: How are these cases usually handled where the variable and imported package name is the same? Just curious why this line compiles without problem but L268 needed to be renamed

Reply:
I think after the variable server is declared, it takes higher priority than the package server. As long as I don't refer to the package server after this line, it compiles.

Reply:
Ah I see. If that's the case I'd prefer just renaming all `server` variable names to prevent ambiguity. Not going to block this PR though, LGTM.